libs/designer-ui/src/lib/settings/settingsection/settingTokenField.tsx

```diff
@@ -212,6 +214,21 @@ const TokenField = ({
     case 'recurrence':
       return <ScheduleEditor readOnly={readOnly} type={editorOptions?.recurrenceType} initialValue={value} onChange={onValueChange} />;
 
+    case 'floatingactionmenu': {
+      const menuItems = getMenuItemsForDynamicAddedParameters(editorOptions?.supportedTypes);
```

Comment:
this function call although reasonable breaks the trend of simply returning JSX components in this `switch case`. Should I structure this code differently?

Reply:
Yes please, this style makes it difficult for the react runtime to do proper optimizaitons.